### PR TITLE
mpich2

### PR DESCRIPTION
**Pull request by ocefpaf**
@bekozi http://www.mpich.org seems to be down, but you take a look at the recipe? Does it looks like what you had in NESII?

### CODE REVIEW

**Comment by msarahan** [comment]
See also: https://github.com/conda-forge/staged-recipes/pull/557

**Comment by ocefpaf** [comment]
> See also: #557

Thanks @msarahan I will try to merge both recipes into one. (I am assuming you don't have to time to finish #557, but you you want I can just close this.)

**Comment by bekozi** [comment]
Looks good! I don't see anything that alarms me. You could consider adding `mpif90 --help` to the test commands...

**Comment by msarahan** [comment]
Go right ahead, @ocefpaf.  I am unlikely to get to it any time soon.

**Comment by ocefpaf** [comment]
@msarahan `mpich` or `mpich2`? I see both names used in different places, not sure what to use here.

**Comment by msarahan** [comment]
IIRC, MPICH is actually MPICH3 - they dropped the versioning in the name.  MPICH is a newer version.

It has been a while since I read about that, though.  Make sure to double-check.

**Comment by bekozi** [comment]
For what it's worth, we use `mpich2` for the `ESMF_COMM` flag.

**Comment by ocefpaf** [comment]
> It has been a while since I read about that, though. Make sure to double-check.

Will do as soon as their site is back online.

> For what it's worth, we use `mpich2` for the `ESMF_COMM` flag.

Does `esmf` needs `mpich2` or is this version OK?

**Comment by bekozi** [comment]
> Does esmf need mpich2 or is this version OK?

It should be okay. I remember there were issues with a bunch of symbols being renamed in newer `mpich` versions, but I thought it was fixed on the `esmf` side. We'll see I guess. I think the build is pretty much the same for earlier versions (`1.4.x` is what the default conda channel uses). So, worst case scenario, we support an earlier version too.

**Comment by ocefpaf** [comment]
> So, worst case scenario, we support an earlier version too.

I can do that later in the feedstock if I get at least one version to work.

**Comment by ocefpaf** [comment]
@msarahan Linux is falinig with:

```shell
libtool: warning: library '/opt/conda/envs/_build/lib/gcc/x86_64-unknown-linux-gnu/4.8.5/../../../libstdc++.la' was moved.
  CCLD     src/env/mpichversion
  CCLD     src/env/mpivars
libtool: warning: library '/opt/conda/envs/_build/lib/gcc/x86_64-unknown-linux-gnu/4.8.5/../../../libgfortran.la' was moved.
/bin/grep: /opt/conda/envs/_build/lib/../lib64/libquadmath.la: No such file or directory
/bin/sed: can't read /opt/conda/envs/_build/lib/../lib64/libquadmath.la: No such file or directory
libtool:   error: '/opt/conda/envs/_build/lib/../lib64/libquadmath.la' is not a valid libtool archive
make[2]: *** [lib/libmpifort.la] Error 1
make[2]: Leaving directory `/opt/conda/conda-bld/work/mpich-3.2'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/opt/conda/conda-bld/work/mpich-3.2'
make: *** [all] Error 2
Command failed: /bin/bash -x -e /opt/conda/conda-bld/work/mpich-3.2/conda_build.sh
```

Fortran support in this library is crucial for @bekozi's `esmf`. Do you have any clue how to proceed here?

File: recipes/mpich/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# FIXME: This is a terrible workaround.
+# Ideally we should fix the information the .la files.
+rm -rf $PREFIX/lib/libquadmath.la
+rm -rf $PREFIX/lib/libgfortran.la
```
Comment (ocefpaf) [inline recipes/mpich/build.sh]: I believe we can avoid this is we don't need the static libs (`--enable-static=no`). @bekozi are they essential?

**Comment by ocefpaf** [comment]
@msarahan the `.la` files are pointing to the original `lib64` directory. (I believe we already saw somewhere, just cannot remember where.) What do you recommend us to do here? Is deleting them at build time OK? Could we patch them (`lib64` &rarr; `lib`)?

**Comment by msarahan** [comment]
Patching them sounds like the right thing to me, if we can't get them right
to begin with. Feel free to take over this pr if you'd like. I am trying to
kick conda build 2.0 out.

**Comment by ocefpaf** [comment]
> Patching them sounds like the right thing to me,

:+1: 

>  if we can't get them right to begin with.

Maybe we will once we have our `gcc` package :wink:

> Feel free to take over this pr if you'd like.

If this you mean `mpich` I already did :smile: 
If you mean `gcc`... No thanks :smiling_imp: 

> I am trying to kick conda build 2.0 out.

Awesome!

**Comment by ocefpaf** [comment]
I cannot get the `Fortran` support to work on `OS X`:

```shell
configure: error: Unable to configure with Fortran support because configure could not determine the size of a Fortran INTEGER.  Consider setting CROSS_F77_SIZEOF_INTEGER to the length in bytes of a Fortran INTEGER
Command failed: /bin/bash -x -e /Users/travis/build/conda-forge/staged-recipes/recipes/mpich/build.sh
```

@bekozi I will skip `OS X` for now and keep trying it in the feedstock.

**Comment by ocefpaf** [comment]
@msarahan if you find some time can you review and merge this one?

@bekozi we don't have `Fortran` support on `OS X` yet, but hopefully we can trigger https://github.com/conda-forge/esmf-feedstock/pull/10 on Linux now.

**Comment by msarahan** [comment]
LGTM.  Thanks for taking this over, @ocefpaf.  Merging.

**Pull request merged**